### PR TITLE
[IMP] core: add method mapped to query object

### PR DESCRIPTION
**Pull request by yelizariev**
this adds more magic to `model._search` which should speed up some searching by
computed field, because we can make a single query instead of getting ids first
and then filtering by that ids

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by yelizariev** [comment]
I can't find button "Make Draft PR"

File: addons/hr_holidays/models/hr_employee.py [hunk 1]
```diff
@@ -150,13 +150,13 @@ def _compute_show_leaves(self):
...
     def _search_absent_employee(self, operator, value):
-        holidays = self.env['hr.leave'].sudo().search([
+        holidays = self.env['hr.leave'].sudo()._search([
             ('employee_id', '!=', False),
             ('state', 'not in', ['cancel', 'refuse']),
             ('date_from', '<=', datetime.datetime.utcnow()),
             ('date_to', '>=', datetime.datetime.utcnow())
         ])
-        return [('id', 'in', holidays.mapped('employee_id').ids)]
+        return [('id', 'in', holidays.mapped('employee_id'))]
```
Comment (rco-odoo) [inline addons/hr_holidays/models/hr_employee.py:159]: You don't need `mapped`.  The existing API fits the operator `inselect`:
```suggestion
        return [('id', 'inselect', holidays.select('"hr_leave"."employee_id"'))]
```

Comment (rco-odoo) [reply addons/hr_holidays/models/hr_employee.py:159]: Note that we could improve `select` to automatically namespace a single identifier, so you could simply do `query.select('employee_id')`.

**Comment by yelizariev** [comment]
So, we have

``` py
    def _search_absent_employee(self, operator, value):
        holidays = self.env['hr.leave'].sudo()._search([...])
```
and then

``` py
        return [('id', 'inselect', holidays.select('employee_id'))]
```
vs

``` py
        return [('id', 'in', holidays.mapped('employee_id'))]
```

There are just two usage of ``inselect`` in Odoo community v14:

https://github.com/odoo/odoo/blob/4de0ae434d1adc97fc1077b8d0f23c7e800ec161/addons/sale_project/models/project.py#L124

https://github.com/odoo/odoo/blob/4de0ae434d1adc97fc1077b8d0f23c7e800ec161/addons/website_forum/models/forum.py#L368

Are we sure that we want more broad usage of ``inselect`` (add it to docs, etc.) or keep it hidden, low-level feature? 

@odony ?

**Pull request closed without merging**